gamedata.fbs

```diff
@@ -0,0 +1,367 @@
+  /// Number of own-goals scored.
+  own_goals:uint;
+  /// Number of goals assisted.
+  assists:uint;
+  /// Number of shots saved.
+  saves:uint;
+  /// Number of shots on opponent goal.
+  shots:uint;
+  /// Number of demolitions made.
+  demolitions:uint;
+}
+
+/// The physical state of an object.
+struct Physics {
+  location:Vector3;
+  rotation:Rotator;
+  velocity:Vector3;
+  angular_velocity:Vector3;
+}
+
+/// Possible states of a car in regards to ground contact and jump/dodging forces.
+/// See more about jumping physics at https://wiki.rlbot.org/botmaking/jumping-physics/
+enum AirState: ubyte {
+  /// All wheels are on the ground and the car is affected by wall-stickiness forces.
+  OnGround,
+  /// The car is currently affected by jumping forces of an initial jump.
+  /// Lasts until the player lets go of jump button but at most for 0.2 seconds (240 ticks).
+  /// The following AirState is typically InAir.
+  Jumping,
+  /// The car is currently affected by jumping forces of a secondary jump (just an impulse in practice).
+  /// Lasts for 13 ticks.
+  /// The following AirState is typically InAir.
+  DoubleJumping,
+  /// The car is currently affected by forces and torque of a dodges.
+  /// This lasts for 79 ticks.
+  /// The following AirState is typically InAir.
+  Dodging,
+  /// The car is free falling.
+  InAir,
+}
+
+/// A collection of information about a player and their car.
+table PlayerInfo {
+  /// The physical state of the player's car.
+  physics:Physics (required);
+  /// The various scores of this player, e.g. those on the leaderboard.
+  score_info:ScoreInfo (required);
+  /// The collision shape of the player's car.
+  hitbox:BoxShape (required);
+  /// The center of the collision shape in local coordinates.
+  // TODO: Is the hitbox tilted? If so, by how much?
+  hitbox_offset:Vector3 (required);
+  /// Information about the latest touch with a ball.
+  /// Is null if the player has yet to touch the ball.
+  latest_touch:Touch;
+  /// Whether the player's car is on the ground or in the air, and what jump/dodging forces current affects the car.
+  air_state:AirState;
+  /// How long until the player cannot dodge anymore. Is -1 while on ground or when airborne for too long after jumping.
+  dodge_timeout:float;
+  /// How long until the player is not demolished anymore. Is -1 if not demolished.
+  demolished_timeout:float;
+  /// Whether the player's car is moving at supersonic speed and can demolish.
+  is_supersonic:bool;
+  /// Wehther the player is a bot or a human.
+  is_bot:bool;
+  /// The name of the player as it appears in game, i.e. possibly appended with "(2)".
+  /// The original name can be found in the match configuration.
+  name:string (required);
+  /// The team of the player.
+  team:uint;
+  /// The current boost.
+  boost:uint;
+  /// The spawn id of the player.
+  /// This value is mostly used internally to keep track of which car was spawned for which client.
+  // TODO: Is this correct Virx?
```

Comment:
Yep, it's correct. I might add that it's guaranteed to be a unique ID for this player throughout the duration of the current match.

Reply:
Okay! I still don't understand why it will ever be useful to a user, but let's keep that discussion to https://github.com/RLBot/core/issues/57.

Reply:
A spawn value of 0 has a special meaning in the PlayerConfiguration, right? Can you explain that?

Reply:
And if my description is correct, why does ScriptConfigurations have a spawn id?

Reply:
> A spawn value of 0 has a special meaning in the PlayerConfiguration, right? Can you explain that?

0 is the spawn id of a human player spawned by RLBot. All future human players will have a unique spawn id.

Reply:
> Okay! I still don't understand why it will ever be useful to a user, but let's keep that discussion to [RLBot/core#57](https://github.com/RLBot/core/issues/57).

It's a nicety for some developers. I know `rlgym_compat` relies on it. It just makes life a bit easier in some niche situations.